crates/polars-ops/src/series/ops/arg_min_max.rs

```diff
@@ -195,24 +212,32 @@ fn first_unset_bit(mask: &Bitmap) -> usize {
 }
 
 fn arg_min_str(ca: &Utf8Chunked) -> Option<usize> {
+    if ca.is_empty() || ca.null_count() == ca.len() {
+        return None;
+    }
     match ca.is_sorted_flag() {
-        IsSorted::Ascending => Some(0),
-        IsSorted::Descending => Some(ca.len() - 1),
+        IsSorted::Ascending => ca.first_non_null(),
+        IsSorted::Descending => ca.last_non_null(),
         IsSorted::Not => ca
             .into_iter()
             .enumerate()
+            .flat_map(|(idx, val)| val.map(|val| (idx, val)))
```

Comment:
Maybe I should remove this and perform a `match` in `reduce`?

Reply:
Match in reduce will be faster as we have less indirection. I think we should first loop over `downcast_iter` and then loop over the array.

Reply:
> Match in reduce will be faster as we have less indirection.
 Make sense, will rewrite this.

> I think we should first loop over downcast_iter and then loop over the array.
Does this means rewriting it to the same pattern of `arg_max_numeric` 🤔

Reply:
They are similar. We could even share a same generic if you are up for that. ^^

But that is maybe a nice follow up PR. :)

Reply:
I am trying to do this refactoring: Let them both share a same generic function as they are almost similar. The only difference is that for numerical types, we have a fast path with the following bound:

https://github.com/pola-rs/polars/blob/ecb819a6541cbc53bee1bf100d2abc87bb9d0962/crates/polars-ops/src/series/ops/arg_min_max.rs#L248

But for `Utf8ChunkedArray`: we do not have this bound, also we doesn't have this branch:

https://github.com/pola-rs/polars/blob/ecb819a6541cbc53bee1bf100d2abc87bb9d0962/crates/polars-ops/src/series/ops/arg_min_max.rs#L264-L268

I can think of some hack solution (such as a magical macro), but I don't really like this way. Do we have a good solution for this that looks simpler and cleaner 🤔.